3.12.1 compilation broken

PR #43 fails:
```
$ make
ocamlbuild -r \
  -tag debug \
  -I . -I ocamltop -I emacs -I vim -I sublime -I gedit \
  -use-ocamlfind -pkg unix -pkg re -pkg re.pcre -pkg cmdliner \
  -pp /home/rjbou/ocamlpro/opam-user-setup/pp_401.ml \
  ousMain.native
+ /home/rjbou/.opam/3.12.1/bin/ocamlfind ocamlopt -c -g -package cmdliner -package re -package re.pcre -package unix -pp /home/rjbou/ocamlpro/opam-user-setup/pp_401.ml -I vim -I gedit -I emacs -I ocamltop -I sublime -o ousMisc.cmx ousMisc.ml
/tmp/camlasm080479.s: Assembler messages:
/tmp/camlasm080479.s:218: Error: junk `revapply (%rip)' after expression
/tmp/camlasm080479.s:218: Error: missing or invalid immediate expression `25revapply@GOTPCREL(%rip)'
/tmp/camlasm080479.s:233: Error: junk `apply (%rip)' after expression
/tmp/camlasm080479.s:233: Error: missing or invalid immediate expression `25apply@GOTPCREL(%rip)'
/tmp/camlasm080479.s:337: Error: junk `apply (%rip)' after expression
/tmp/camlasm080479.s:337: Error: missing or invalid immediate expression `25apply@GOTPCREL(%rip)'
/tmp/camlasm080479.s:379: Error: junk `apply (%rip)' after expression
/tmp/camlasm080479.s:379: Error: missing or invalid immediate expression `25apply@GOTPCREL(%rip)'
File "/tmp/camlpp2422ae", line 1, characters 0-1:
Error: Assembler error, input left in file /tmp/camlasm080479.s
Command exited with code 2.
Compilation unsuccessful after building 21 targets (0 cached) in 00:00:00.
Makefile:19: recipe for target '_build/ousMain.native' failed
make: *** [_build/ousMain.native] Error 10
```
context:
```
$ opam config report
# opam config report
# opam-version      2.0.0~rc2 (a1737429b7b20c43cc24e52670bb6809c5505fb2) 
# self-upgrade      no
# system            arch=x86_64 os=linux os-distribution=debian os-version=9.4
# solver            builtin-mccs+glpk
# install-criteria  -removed,-count[version-lag,request],-count[version-lag,changed],-changed
# upgrade-criteria  -removed,-count[version-lag,solution],-new
# jobs              7
# repositories      1 (http), 1 (local)
# pinned            0
# current-switch    3.12.1
```
and
```
$ opam list
# Packages matching: installed
# Name              # Installed # Synopsis
base-bigarray       base        Bigarray library distributed with the OCaml compiler
base-bytes          backport    Bytes compatibility library distributed with ocamlfind
base-ocamlbuild     base        OCamlbuild binary and libraries distributed with the OCaml compiler
base-threads        base        Threads library distributed with the OCaml compiler
base-unix           base        Unix library distributed with the OCaml compiler
cmdliner            0.9.8       Declarative definition of command line interfaces for OCaml
conf-m4             1           Virtual package relying on m4
ocaml               3.12.1      The OCaml compiler (virtual package)
ocaml-base-compiler 3.12.1      Official 3.12.1 release
ocamlbuild          0           Build system distributed with the OCaml compiler since OCaml 3.10.0
ocamlfind           1.7.3-1     A library manager for OCaml
re                  1.7.1       RE is a regular expression library for OCaml
```

--- a/opam
+++ b/opam
@@ -3,10 +3,10 @@ name: "user-setup"
 version: "0.7"
 maintainer: "Louis Gesbert <louis.gesbert@ocamlpro.com>"
 authors: "Louis Gesbert <louis.gesbert@ocamlpro.com>"
-homepage: "https://github.com/AltGr/opam-user-setup"
-bug-reports: "https://github.com/AltGr/opam-user-setup/issues"
+homepage: "https://github.com/OCamlPro/opam-user-setup"
+bug-reports: "https://github.com/OCamlPro/opam-user-setup/issues"
 license: "ISC"
-dev-repo: "https://github.com/AltGr/opam-user-setup.git"
+dev-repo: "https://github.com/OCamlPro/opam-user-setup.git"
 build: [make]
 depends: [
   "ocamlfind" {build}

--- a/ousMisc.ml
+++ b/ousMisc.ml
@@ -15,7 +15,9 @@ type 'a stringmap = 'a StringMap.t
 
 let lines_of_string s =
   let rex = Re.(compile (char '\n')) in
-  Re_pcre.split ~rex (String.trim s)
+  let s = Re.(replace_string (compile @@ seq [ bos; rep space]) "" s) in
+  let s = Re.(replace_string (compile @@ seq [ rep space; eos]) "" s) in
+  Re_pcre.split ~rex s
 
 let lines_of_channel ic =
   let rec aux acc =

--- a/ousMisc.ml
+++ b/ousMisc.ml
@@ -2,9 +2,8 @@ open OusTypes
 
 let msg fmt = Printf.kprintf print_endline fmt
 
-external (@@) : ('a -> 'b) -> 'a -> 'b = "%apply"
-
-external (|>) : 'a -> ('a -> 'b) -> 'b = "%revapply"
+let (|>) a f = f a
+let (@@) f a = f a
 
 let (/) = Filename.concat
 

